arangod/RocksDBEngine/RocksDBZkdIndex.cpp

```diff
@@ -763,16 +788,32 @@ Result RocksDBZkdIndex::remove(transaction::Methods& trx,
                                velocypack::Slice doc,
                                OperationOptions const& /*options*/) {
   TRI_ASSERT(_unique == false);
-  TRI_ASSERT(_sparse == false);
 
-  auto keyValue = readDocumentKey(doc, _fields);
+  zkd::byte_string keyValue;
+  {
+    auto result = readDocumentKey(doc, _fields);
+    if (result.fail()) {
+      if (result.errorNumber() == TRI_ERROR_QUERY_INVALID_ARITHMETIC_VALUE &&
+          _sparse) {
+        return {};
+      }
+      THROW_ARANGO_EXCEPTION(result.result());
+    }
+    keyValue = std::move(result.get());
+  }
 
   RocksDBKey rocksdbKey;
   uint64_t hash = 0;
   if (!isPrefixed()) {
     rocksdbKey.constructZkdIndexValue(objectId(), keyValue, documentId);
   } else {
-    auto prefixValues = extractAttributeValues(trx, _prefixFields, doc);
+    auto result = extractAttributeValues(trx, _prefixFields, doc, !_sparse);
```

Comment:
The new parameter `nullAllowed` is now set to true if `!_sparse`, i.e. we allow null values if the index is not sparse.
Shouldn't this be opposite, i.e. 
```suggestion
    auto result = extractAttributeValues(trx, _prefixFields, doc, _sparse);
```
?